handling-network-upgrades: Describe seamless/dump&restore upgrades (#96)
oasisprotocol/docs: docs/general/foundation/testnet/README.md

* [Oasis Core](https://github.com/oasisprotocol/oasis-core) version:
  * [22.1.5](https://github.com/oasisprotocol/oasis-core/releases/tag/v22.1.5)
  * To sync your node from the genesis, use the following Oasis

>     Core version first ([read more](upgrade-log.md#2022-04-04-upgrade)):

Reviewer: The *Upgrade Log* page is quite dense. Would it make sense to create a general page or an entry under Node Operator FAQ for describing how Oasis Core handles syncing from Genesis?

Something along the lines of:

Use the first Oasis Core version to sync blocks from genesis onwards. After reaching the upgrade height _x_, the Oasis Node process will stop with a message similar to the following:
```
log message(s)
```
At that point you need to switch the Oasis Node binary with the one from the next Oasis Core version. After restarting the Oasis Node process, the syncing will resume.

Author: Revamped [handling-network-upgrades.md](https://github.com/oasisprotocol/docs/blob/matevz/feature/testnet-upgrade-info/docs/general/run-a-node/maintenance-guides/handling-network-upgrades.md) to include this.

>   * To sync with genesis document, perform a seamless upgrade from the previous
>     version first ([read more][handling network upgrades]):

Reviewer 2: ```suggestion
  * To sync from genesis, you need to start with an earlier version first
    ([read more][handling network upgrades]):
```

:::

> [handling network upgrades]: ../../run-a-node/maintenance-guides/handling-network-upgrades.md

Reviewer 2: Hm the link in question doesn't seem to answer the question on what to do when syncing from genesis. There should be a special section on that page called "Syncing from genesis" (or similar) explicitly explaining what to do in this case (e.g. when there have been _multiple seamless upgrades_ since genesis) even if it seems obvious.

Use an example where two (or three) versions are specified in network parameters and explain the steps where the earliest version is downloaded first and syncing is started using that one until the node stops for upgrade by itself. Then continue with the next version until the same happens and finally use the latest specified version.

Author: Revamped the beginning so that it also puts syncing from genesis into context.
